core/src/main/kotlin/fr/sncf/osrd/stdcm/graph/PostProcessingSimulation.kt

```diff
@@ -198,11 +199,28 @@ private fun initFixedPoints(
         )
         prevStopTime += stop.duration
     }
+
+    // Add one point at the end to match the standard allowance (if any)
     if (hasStandardAllowance && res.none { it.offset == length })
         res.add(makeFixedPoint(res, edges, length, length, updatedTimeData, 0.0))
 
     // Add points at the end of each engineering allowance
-    fun addFixedPointAvoidingDuplicates(offset: Distance) {
+    var prevEdgeLength = 0.meters
```

Comment:
How about renaming it to:
```suggestion
    var edgeStartOffset = 0.meters
```